File: ios/Classes/events/AbstractMappableEvent.swift
//
//  AbstractMappableEvent.swift
//  Runner
//
//  Created by Sweet Pea on 22/11/2023.
//

import Foundation


class AbstractMappableEvent<T : IMappable> : AbstractEvent {
    
    func raise(_ payload: T) {
        super.raise(payload.toMap)
    }
}

File: sabian_native_common_android/android/src/main/kotlin/com/sabiantech/sabian_native_common_android/events/AbstractMappableEvent.kt
package com.sabiantech.sabian_native_common_android.events

import com.sabiantech.sabian_native_common_android.structures.IMappable

abstract class AbstractMappableEvent<T : IMappable> : AbstractEvent() {
    fun raise(payload: T) {
        super.raise(payload.toMap)
    }
}